presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestingPrestoClient.java

```diff
@@ -85,6 +85,11 @@ public T execute(@Language("SQL") String sql)
 
     public T execute(Session session, @Language("SQL") String sql)
     {
+        return executeWithId(session, sql).getResult();
+    }
+
+    public ResultWithQueryId<T> executeWithId(Session session, @Language("SQL") String sql)
```

Comment:
I only count 7 usages in total of AbstractTestingPrestoClient.execute, so it shouldn't be that big of a change, no?

I think it can be fixed by just doing 2) from your list above.